GPU/Common/PresentationCommon.cpp

```diff
@@ -233,14 +233,21 @@ bool PresentationCommon::UpdatePostShader() {
 
 	if (gstate_c.Use(GPU_USE_SIMPLE_STEREO_PERSPECTIVE)) {
 		const ShaderInfo *stereoShaderInfo = GetPostShaderInfo(g_Config.sStereoToMonoShader);
-		bool result = CompilePostShader(stereoShaderInfo, &stereoPipeline_);
-		if (!result) {
+		if (stereoShaderInfo) {
+			bool result = CompilePostShader(stereoShaderInfo, &stereoPipeline_);
+			if (!result) {
+				// We won't have a stereo shader. We have to check for this later.
+				delete stereoShaderInfo_;
+				stereoShaderInfo_ = nullptr;
+				stereoPipeline_ = nullptr;
+			} else {
+				stereoShaderInfo_ = new ShaderInfo(*stereoShaderInfo);
```

Comment:
Shouldn't we still delete the old one beforehand?  Or if `DestroyStereoShader();` does it, why do we need to delete in these other two paths?

-[Unknown]

Reply:
I dunno, I think I confused myself. I'll clean it up.